Backend/Schedules/Service/AppointmentService.cs

```diff
@@ -1,156 +1,91 @@
+            foreach(Appointment appointmentIt in allAppointments)
+            {
+                Appointment appointment = occupied.FirstOrDefault(a => a.isOccupied(appointmentIt.Start, appointmentIt.End));
+                if (isOccupied(appointment))
+                    available.Remove(appointmentIt);
+            }
+            return available;
         }
-        public int GetNumberOfAppointmentsFor(Doctor doctor, TypeOfAppointment type)
+        private bool isOccupied(Appointment appointment)
         {
-            var allExams = appointmentRepository.GetAll().ToList().Where(entity => entity.IsDoctor(doctor));
-            return allExams.Where(entity => entity.Finished == true && entity.TypeOfAppointment == type).ToList().Count;
+            return appointment != null && !appointment.canceledByPatient;
         }
-
-        public Appointment FinishAppointment(Appointment appointment)
+    
+        public List<Appointment> GetByDoctorAndDate(string doctorId, DateTime date)
         {
-            appointment.Finished = true;
-            return appointmentRepository.Update(appointment);
+            return _appointmentRepository.GetByDoctorIdAndDate(doctorId, date).ToList();
         }
 
-        public Appointment GetCurrentAppointment(Doctor doctor, MedicalRecord medicalRecord)
-        {
-            var doctorsDay = GetScheduledBy(doctor, DateTime.Today);
-            return doctorsDay.SingleOrDefault(entity => entity.IsMedicalRecord(medicalRecord));
-        }
-        public Appointment ChangeDoctorForAppointment(Doctor doctor, Appointment appointment)
-        {
-            Appointment appointmentToUpdate = appointmentRepository.GetObject(appointment.Id);
-            appointmentToUpdate.Doctor = doctor;
-            return appointmentRepository.Update(appointmentToUpdate);
-        }
-      
-        public Appointment ChangeTimeOfAppointment(Appointment appointment, DateTime termOfAppointment)
+        public List<Appointment> InitializeAppointments(string doctorId, DateTime date)
         {
-            Appointment appointmentToUpdate = appointmentRepository.GetObject(appointment.Id);
-            appointmentToUpdate.Period.StartTime = termOfAppointment;
-            int ifSurgeryMultiply = appointment.TypeOfAppointment == TypeOfAppointment.Surgery ? surgeryPeriod : 1;
-            appointmentToUpdate.Period.EndTime = termOfAppointment.AddMinutes(appointmentTimePeriod * ifSurgeryMultiply);
-            return appointmentRepository.Update(appointmentToUpdate);
-        }
+            List<Appointment> appointments = new List<Appointment>();
+            DoctorWorkDay doctorWorkDays = _doctorWorkDayRepository.GetByDoctorIdAndDate(doctorId, date.Date);
 
-        public bool DeleteAppointment(Appointment appointment) => 
-            appointmentRepository.Delete(appointment);
+            if (doctorWorkDays == null)
+                return appointments;
 
```

Comment:
I think it's better to raise an exception in doctorWorkDayRepo then to compare the return value with null if you don't find any work days. Then the lines 71-72 are not needed.